include/ion-bb-dnn/rt_tvm.h

```diff
@@ -0,0 +1,520 @@
+
+    assert(array->ndim == indices.size());
+    // TODO: Support strides.
+    assert(array->strides == nullptr);
+    // TODO: Support offset.
+    assert(array->byte_offset == 0U);
+
+    int64_t *shape = array->shape;
+    int64_t index = 0;
+    for (auto dim = decltype(indices.size())(0); dim < indices.size(); ++dim) {
+        int64_t stride = 1;
+        for (size_t i = dim + 1; i < array->ndim; ++i) {
+            stride *= shape[i];
+        }
+
+        index += indices[dim] * stride;
+    };
+
+    return reinterpret_cast<T *>(array->data)[index];
+}
+
+std::vector<DetectionBox> postprocess(TVMArrayHandle nums,
+                                      TVMArrayHandle boxes,
+                                      TVMArrayHandle labels,
+                                      TVMArrayHandle scores) {
+    // Assume single batch.
+    const int64_t batch = 0;
+    int32_t num = -1;
+    auto dtype_code = nums->dtype.code;
+    if (dtype_code == kDLInt) {
+        num = (int32_t)ndarray_at<int32_t>(nums, {0});
+    } else if (dtype_code == kDLFloat) {
+        num = (int32_t)ndarray_at<float>(nums, {0});
+    }
+
+    if (num < 0) {
+        std::cerr << "Error: failed to get num" << std::endl;
+    }
+
+    std::vector<DetectionBox> valid_boxes;
+    for (auto i = decltype(num)(0); i < num; ++i) {
+        // Added offset of background.
+        const auto class_id =
+            static_cast<int32_t>(ndarray_at<float>(labels, {batch, i})) + 1;
+        const auto confidence = ndarray_at<float>(scores, {batch, i});
+
+        // Boxes is formatted as [y1, x1, y2, x2] nomalized [0, 1)
+        const auto x1 = ndarray_at<float>(boxes, {batch, i, 1});
+        const auto y1 = ndarray_at<float>(boxes, {batch, i, 0});
+        const auto x2 = ndarray_at<float>(boxes, {batch, i, 3});
+        const auto y2 = ndarray_at<float>(boxes, {batch, i, 2});
+
+        valid_boxes.emplace_back(
+            DetectionBox{class_id, confidence, x1, x2, y1, y2});
+    }
+
+    return valid_boxes;
+}
+
+}  // namespace
+
+class TVMSessionManager {
+private:
+    bool init_configs_ = false;
+    DLContext dlctx_;
+    std::string model_base_url_ = "";
+    std::string model_cache_dir_ = "";
+    bool use_cuda_ = false;
+    bool use_edgetpu_ = false;
+    std::string dnn_model_name_;
+    std::string target_arch_;
+
+    std::unordered_map<std::string, TVMModuleHandle> modules_;
+    TVMFunctionHandle runtime_creator_;
+
+    bool is_available_tvm_;
+    TVMSessionManager()
+        : is_available_tvm_(false) {
+        if (!tvm_runtime_init()) {
+            std::cerr << "Log: could not init the tvm runtime" << std::endl;
+            return;
+        }
+        is_available_tvm_ = true;
+    }
+
+    void init_runtime_creator() {
+
+        if (use_edgetpu_) {
+            TVMFuncGetGlobal("tvm.edgetpu_runtime.create", &runtime_creator_);
+        } else {
+            TVMFuncGetGlobal("tvm.graph_runtime.create", &runtime_creator_);
```

Comment:
@tommyhuangtw Fix the key to `tvm.graph_executor.create` , so it may works.
`graph_runtime` has renamed to `graph_executor` in the below PR.
https://github.com/apache/tvm/pull/7653

Reply:
@akmaru It works now. Thank you very much!